Source/src/ui/WindowResource.cpp

```diff
@@ -7,73 +7,113 @@
     {
-        current_directory = current_directory.parent_path();
-    }    
+        if (ImGui::Button(ICON_FA_ARROW_LEFT, {thumbnail_size * 0.5f, thumbnail_size * 0.3f}))
+        {
+            current_path = current_path.parent_path();
+            updated = false;
+            ImGui::End();
+            return;
+        }
+        ImGui::NextColumn();
+    }
 
-    for (auto& directory_entry : std::filesystem::directory_iterator(current_directory))
-    {
-        const auto& path = directory_entry.path();
-        auto relative_path = relative(path, assets_path);
-        std::string filename = relative_path.filename().string();
+    CreateMaterialGui({thumbnail_size, thumbnail_size * 0.3f});
 
-        if (directory_entry.is_directory())
+    for (const auto& directory_entry : current_content)
+    {
+        if (directory_entry.is_directory)
         {
-            if (ImGui::Button(filename.c_str()))
+            if (ImGui::Button(StringUtils::Concat(ICON_FA_FOLDER, " ", directory_entry.name).c_str(), {thumbnail_size, thumbnail_size * 0.3f}))
             {
-                current_directory /= path.filename();
+                current_path = directory_entry.path;
+                updated = false;
+                break;
             }
+            ImGui::NextColumn();
         }
-        else
+    }
+
+    ImGui::Columns(MAX_COLUMN_FILES);
+    ImGui::Separator();
+
+    int items_counter = 0;
+
+    for (const auto& directory_entry : current_content)
+    {
+        if (!directory_entry.is_directory)
         {
-            if (FileSystem::GetFileExtension(filename.c_str()) != META_EXTENSION)
+            if (FileSystem::GetFileExtension(directory_entry.name.c_str()) == META_EXTENSION)
+            {
+                continue;
+            }
+            
+            auto selection = ImGui::Selectable(directory_entry.name.c_str(), false, ImGuiSelectableFlags_AllowDoubleClick);
+            if (ImGui::IsMouseDoubleClicked(ImGuiMouseButton_Left) && ImGui::IsItemHovered())
             {
-                auto selection = ImGui::Selectable(filename.c_str(), ImGuiSelectableFlags_AllowDoubleClick);
-                if (ImGui::IsMouseDoubleClicked(selection) && ImGui::IsItemHovered())
-                {
-                    filename.insert(0, "\\");
-                    filename.insert(0, current_directory.string().c_str());
-                    LoadAsset(filename);
-                    App->event->Publish(Event::Type::CREATE_EDITOR_HISTORY_ENTRY);
-                }
-            }            
+                LoadAsset(directory_entry.path.string());
+                App->event->Publish(Event::Type::CREATE_EDITOR_HISTORY_ENTRY);
+            }
+
+            ++items_counter;
+
+            if (items_counter > MAX_FILES_PER_COLUMN)
+            {
+                ImGui::NextColumn();
+                items_counter = 0;
+            }
         }
     }
+
+    ImGui::Columns(1);
```

Comment:
Columns is the old API way of organizing content, the new one is BeginTable, see Widgets::DragFloat for an example